### DISCUSSION TITLE
Host trivyscanner on Azure Container Instance and run it to scan an image hosted on Azure Container Registry

### DISCUSSION THREAD

**Discussion by pwalecha** [aquasecurity/trivy#5231] [Q&A]
### Question

I am trying to run a scenario where I want to host the Trivy Scanner on Azure Container Instance. Using that I want to run a Trivy scan for vulnerability and generate SBOM on the image hosted on my private azure container registry.

Question is, I would want to run the Trivy scanner using User Identity that has permission (AcrPull) on the Azure Container Registry. How do I pass that information within the Trivy Command. I saw documentation mentioned in 
https://aquasecurity.github.io/trivy/v0.45/docs/advanced/private-registries/acr/
However, I don't want to deal with Client ID and Secret. Can I used Managed User Identity instead ? If yes, how ?  

### Target

Container Image

### Scanner

Vulnerability

### Output Format

JSON

### Mode

Standalone

### Operating System

Linux

### Version

```bash
2
```

**Comment by itaysk** [comment]
I think this is more of an Azure question than a Trivy question. You can try to follow this guide, which ends up with `az azr login` command to do the equivalent of `docker login` but with the MSI. https://learn.microsoft.com/en-us/azure/container-registry/container-registry-authentication-managed-identity?tabs=azure-cli#use-the-identity-to-access-the-registry-1

**Reply by pwalecha** [reply]
I'll re-frame the question a bit, what values as a KEY NAME I should be passing in as the ENVIRONMENT VARIABLES that TRIVY uses, for the managed identity? like for Service Principal, and Secrets, I expects: AZURE_CLIENT_ID,AZURE_CLIENT_SECRET and AZURE_TENANT_ID. If I intend to use managed identity, what equivalent should be passed as the key name.

**Reply by shankarvrp** [reply]
@pwalecha - Did you find a way to this? I am looking to run Trivy with Azure Managed Identity

**Reply by chlynott** [reply]
I would also like to use a managed Identity that we already have with Acr Pull permissions. Any guidance would be appreciated, as we are not currently using workload identity in our clusters and do not plan too currently.
I am also using the Trivy operator helm chart, so having an option for this value would be great
